### DISCUSSION TITLE
accountability is missing

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#6535] [Archive]
I see now the `accountability` key is missing and I was using it to detect if the API calls come from the browser webhook or from direct API call update by using a custom user agent. How can we achieve this feature in the latest `81` version?

**Comment by rijkvanzanten** [comment]
The user agent has been stripped from the webhook payload for privacy and security reasons. The user/role are still available in the webhook payload:

https://github.com/directus/directus/blob/043ae13e40291beed50615329967f13ee241025d/api/src/webhooks.ts#L48-L56

**Reply by botzill** [reply]
I see, but user and role are not enough to understand where the requests come from. Maybe we can have separate fields that tell us if this request is from the interface webhook? Would really helpful to have this.

**Reply by botzill** [reply]
Also, would be useful to have an option on API calls that we can disable the call to webhook. Do you think this is smth useful ?

**Reply by rijkvanzanten** [reply]
Couldn't you simply add a query param to your URL? Eg configure the webhook with url = `https://example.com?ref=directus`

**Reply by botzill** [reply]
This will not help. When I update some entries from the browser, webhook is called with that param.  When I update some entries from nodejs, webhook is called again and will have same params. We need a way to differentiate these 2 types of calls. Right now added a new field `api_call` which is hidden and I set it when  I update via nodejs, and I know if I have this field it comes from nodejs update.

**Comment by philleepflorence** [comment]
Hi, hope this helps with your question,

### Active Fingerprinting and Identification
As most browser vendors are moving towards disabling passive fingerprinting ([UA Detailed Strings](https://css-tricks.com/freezing-user-agent-strings/) and [Referrer Header Information](https://developers.google.com/web/updates/2020/07/referrer-policy-new-chrome-default)), in addition to the available webhook payload, I personally also use URL query variables to differentiate App and API Webhooks Requests, if I manage the WebHook Response as well.

**Reply by botzill** [reply]
I think having smth nativ for this is a good idea, as we all have to do this custom field or URL param to achieve it.
